index.js

```diff
@@ -65,3 +67,125 @@ module.exports = (emitter, event, options) => {
 
 	return ret;
 };
+module.exports.iterator = (emitter, event, options) => {
+	if (typeof options === 'function') {
+		options = {filter: options};
+	}
+
+	options = Object.assign({
+		rejectionEvents: ['error'],
+		resolutionEvents: [],
```

Comment:
Should we have any default names here, like `['end', 'finish']`?

Reply:
I wouldn't.  I think the default for error is a special case, since `node` will throw a fit if you don't handle 'error' events.